### PR TITLE
Simulated Test for Friendly Kickoff Play

### PR DESCRIPTION
**Pull request by Resocram**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp:53]: We should also check that the ball is kicked right?

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: Could this be replaced by `robotsInFriendlyHalf` implemented here https://github.com/UBC-Thunderbots/Software/pull/1946/files?

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (Resocram) [reply src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: Yes I believe it can, I'll draft this PR for now until we merge that PR, I rather not have duplicate validation functions

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (Resocram) [reply src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp:53]: Added

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (Resocram) [reply src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: Refactored with new validation functions from https://github.com/UBC-Thunderbots/Software/pull/1946/files

File: src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.h
Comment (kelvinkoon) [inline src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.h]: Nit: Is there a reason `robot` isn't included in the `@param` list? I noticed `robotID` is included in the `@param` list below so I'm wondering if there's an inconsistency.

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
Quick nit, LGTM otherwise.

File: src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.h
Comment (Resocram) [reply src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.h]: Added

File: src/software/simulated_tests/terminating_validation_functions/robot_in_center_circle_validation.cpp
Comment (jonathanlew) [inline src/software/simulated_tests/terminating_validation_functions/robot_in_center_circle_validation.cpp]: robotOptional should be snake_case

File: src/software/simulated_tests/terminating_validation_functions/robot_kicked_ball_validation.cpp
Comment (jonathanlew) [inline src/software/simulated_tests/terminating_validation_functions/robot_kicked_ball_validation.cpp]: robotOptional should be snake_case

File: src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.cpp
Comment (jonathanlew) [inline src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.cpp]: The robot here is a copy that won't get updated, so this validation function doesn't work. We need to pass in a robot_id and look it up every time

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: can we use some sort of for loop/ pass a vector of robot ids in?

**Review by jonathanlew — CHANGES_REQUESTED** [review]
good work so far

File: src/software/simulated_tests/terminating_validation_functions/robot_in_center_circle_validation.cpp
Comment (Resocram) [reply src/software/simulated_tests/terminating_validation_functions/robot_in_center_circle_validation.cpp]: Fixed

File: src/software/simulated_tests/terminating_validation_functions/robot_kicked_ball_validation.cpp
Comment (Resocram) [reply src/software/simulated_tests/terminating_validation_functions/robot_kicked_ball_validation.cpp]: Fixed

File: src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.cpp
Comment (Resocram) [reply src/software/simulated_tests/non_terminating_validation_functions/robots_in_friendly_half_validation.cpp]: Fixed, used robot_id instead

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (Resocram) [reply src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]: Fixed

File: src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp
Comment (jonathanlew) [inline src/software/ai/hl/stp/play/kickoff_friendly_play_test.cpp]:
```suggestion
            for (RobotId robot_id : {0, 1, 2, 3, 5})
```

File: src/software/simulated_tests/non_terminating_validation_functions/robots_not_in_center_circle_validation.cpp
Comment (jonathanlew) [inline src/software/simulated_tests/non_terminating_validation_functions/robots_not_in_center_circle_validation.cpp]: why can't this use robotNotInCenterCircle?

Comment (Resocram) [reply src/software/simulated_tests/non_terminating_validation_functions/robots_not_in_center_circle_validation.cpp]: It can but `robotNotInCenterCircle` takes in a RobotId and finds a robot based on that, but in `robotsNotInCenterCircle`, it already has the robots, so we don't need to find the RobotId. I think it would be redundant if we use `robotNotInCenterCircle` since I would have to get the robot_id from a robot -> pass this into `robotNotInCenterCircle`, when I already have the robot found already.

**Review by jonathanlew — APPROVED** [review]
LGTM!

**Review by kelvinkoon — APPROVED** [review]
LGTM pending CI

**Pull request merged**
